### ISSUE TITLE
AGPL is not recognised by code analysis.

### ISSUE THREAD

**Issue by rotespferd** [thenativeweb/roboter#673]
I use roboter version 12.0.0.

In my package.json the license field looks like this:
`"license": "AGPL-3.0-or-later"`.
The string "AGPL-3.0-or-later" is the SPDX license ID for the GNU Affero General Public License v3.0 or later.

When I run `npx roboter` I get the following result:

```
────────────────────────────────────────────────────────────────────────────────
▸ Running code analysis...
./package.json
  error    Invalid value for license  (license)

✖ 1 problems (1 errors, 0 warnings)

✗ Malformed package.json found.
```

It seams that roboter did not recognise the license as a valid license.

**Comment by goloroden** [comment]
Hey @rotespferd 👋 

Thanks for bringing this up, we are going to investigate what the problem is and may come back with additional questions to you.

**Comment by goloroden** [comment]
Okay, I've tried this with a simple example, and I was not able to reproduce this behavior. Can you please provide a sample repository?

**Comment by rotespferd** [comment]
I created the following package.json, which reproduces the error for me:
https://gist.github.com/rotespferd/f34e703ff67109d225819567c311de91

**Comment by goloroden** [comment]
If I take this `package.json` file, run `npm install`. and then run `npx roboter license`, it works 😳

**Comment by rotespferd** [comment]
This is strange :-D

I looked into the code of roboter. It seams to me that the license check compare the license field of the package.json against the licenses in the file "licenseCheck.json".
The array "compatibleLicenses" in this file does not contain the AGPL. So in my understanding the license check failed because the AGPL is not in the file.

**Comment by goloroden** [comment]
Basically, what we do is that we compare the license given in the `package.json` file against those defined in the `licenseCheck.json` file. If there are any incompatibilities, you will get a license error.

However, the error message you provided does not say that there is an incompatible license, but that the `package.json` file is broken – and this is what I can not reproduce.

**Comment by rotespferd** [comment]
Ok, sorry, that was my fault. I misunderstood the logic of the code.

What happens if you run `npx roboter analyze` with the given package.json?
The problem seems to be in the linting part of the "package.json". roboter uses the module "npm-package-json-lint" for this. "npm-package-json-lint" validates the content of the license field against which licenses are configured in ".npmpackagejsonlintrc" in the property "valid-values-license" (see [documentation](https://npmpackagejsonlint.org/docs/en/rules/valid-values/valid-values-license)).

Roboter extends the config of the file "npm-package-json-lint-config-tnw/app.json", which basically extends "npm-package-json-lint-config-tnw/base.json". In base.json you only allow the MIT license. So every other license produces an error. For testing purpose I inserted the GPL in the license field, which also produces an error. Only when I put the MIT license in the field everything runs without an error.

_(To rule out that the error is due to my environment I tested everything which a freshly installed Linux environment with the latest Node version)_

**Comment by yeldiRium** [comment]
@rotespferd you are entirely correct. The issue here is the npm-package-json-lint configuration which does not allow the license you are using. This is not an issue with the `license` command.

This was originally done in the npm-package-json-lint configuration since we use the MIT license for almost everything, but that has since caused other issues and thus other licenses were added (e.g. your ALPG-3.0-or-later).

If you want to use another license that is not present in the npm-package-json-lint configuration, you can overwrite that rule. You found the documentation on that yourself, so I won't elaborate on this :)

The package `npm-package-json-lint-config-tnw` unfortunately has not yet been updated in the roboter, since we're having CI issues. I'll notify you here once the new version is contained and this issue will be resolved.

**Comment by goloroden** [comment]
@rotespferd Sorry for not having replied back earlier, I had a few days off.

The mistake was on my side – I actually forgot to add a `.npmpackagejsonlintrc` file to the project I have been testing with. If I add one, I get the same error that you got. Sorry for that 😞

As @yeldiRium already pointed out we updated the underlying package accordingly, but have a few problems releasing a new version of roboter. As soon as this is done the problems should be gone.

**Comment by yeldiRium** [comment]
@rotespferd this issue is resolved in roboter version `12.1.0` with the update of the dependency `npm-package-json-lint-config-tnw` to version `1.1.0`. If in the future you choose to use a license that is not intended to be used by the config, you need to override that rule yourself.

**Comment by rotespferd** [comment]
Thank you guys for your work!
Now it is working for me.
